File: ios/hybrids/NitroPlayer/HybridNitroPlayerFactory.swift
//
//  HybridNitroPlayerFactory.swift
//  NitroPlay
//
//  Created by Krzysztof Moch on 09/10/2024.
//

import Foundation
import NitroModules

class HybridNitroPlayerFactory: HybridNitroPlayerFactorySpec {
  func createPlayer(source: HybridNitroPlayerSourceSpec) throws -> HybridNitroPlayerSpec {
    return try HybridNitroPlayer(source: source)
  }
}

File: android/src/main/java/com/nitroplay/video/hybrids/videoplayer/HybridNitroPlayerFactory.kt
package com.margelo.nitro.video

import androidx.annotation.OptIn
import androidx.media3.common.util.UnstableApi
import com.facebook.proguard.annotations.DoNotStrip

@DoNotStrip
class HybridNitroPlayerFactory(): HybridNitroPlayerFactorySpec() {
  @OptIn(UnstableApi::class)
  override fun createPlayer(source: HybridNitroPlayerSourceSpec): HybridNitroPlayerSpec {
    return HybridNitroPlayer(source as HybridNitroPlayerSource)
  }

  override val memorySize: Long
    get() = 0
}
